common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -1156,6 +1157,7 @@ def _update_module_location(module, new_location):
             module.get_explicitly_set_fields_by_scope(Scope.content).keys() +
             module.get_explicitly_set_fields_by_scope(Scope.settings).keys()
         )
+        original_module = copy.deepcopy(module)
```

Comment:
I also have performance concerns with the `copy.deepcopy`. You could avoid the deepcopy by looping through the rekey fields twice below - one before setting the new location to obtain the old keys/values -and- one after setting the new location to delete the old keys and set the new keys/values. Admittedly, I don't have any hard data to back up these performance concerns - but seems like an OK tradeoff for avoiding the deepcopy.